### PR TITLE
Datetime filter fix

### PR DESCRIPTION
**Pull request by WesleyPumpkinhead**
@luke The datetime filter had been changed to no longer accept date strings, which broke certain templates, so I have added this functionality back in

### CODE REVIEW

File: src/core/javascripts/filters/main.js.coffee [hunk 1]
```diff
@@ -295,10 +294,15 @@ angular.module('BB.Filters').filter 'datetime', (SettingsService) ->
       uk: 'HH:mm'
 
   (date, format="LLL", show_time_zone=false) ->
+
     if hardcoded_formats[format]
       cc = if SettingsService.getCountryCode() is 'us' then 'us' else 'uk'
       format = hardcoded_formats[format][cc]
 
+    # if date is not a moment object, convert it to a moment object
+    if date and !moment.isMoment(date)
```
Comment (lukeferguson) [inline src/core/javascripts/filters/main.js.coffee]: I would argue that that it's the responsibility of the caller to ensure a moment object is passed in, however, if this is still nessecary, a validity check would be appropriate too, see http://momentjs.com/docs/#/parsing/is-valid/

**Comment by WesleyPumpkinhead** [comment]
@lukeferguson Moment validation added

File: src/core/javascripts/filters/main.js.coffee [hunk 2]
```diff
@@ -295,10 +294,15 @@ angular.module('BB.Filters').filter 'datetime', (SettingsService) ->
...
+      date = moment(date) if moment(date).isValid()
+
     if date and moment.isMoment(date)
```
Comment (lukeferguson) [inline src/core/javascripts/filters/main.js.coffee]: replace `if date and moment.isMoment(date)` with if date and date.isValid()

File: src/core/javascripts/filters/main.js.coffee [hunk 3]
```diff
@@ -295,10 +294,15 @@ angular.module('BB.Filters').filter 'datetime', (SettingsService) ->
...
 
   (date, format="LLL", show_time_zone=false) ->
+
     if hardcoded_formats[format]
       cc = if SettingsService.getCountryCode() is 'us' then 'us' else 'uk'
       format = hardcoded_formats[format][cc]
 
+    # if date is not a moment object, convert it to a moment object
+    if date and !moment.isMoment(date)
+      date = moment(date) if moment(date).isValid()
```
Comment (lukeferguson) [inline src/core/javascripts/filters/main.js.coffee]: Do the validity check later as proposed in my other line note

**Comment by WesleyPumpkinhead** [comment]
@lukeferguson moment validation updated as per your request

**Pull request merged**